### PR TITLE
Add relevant Metadata fields to Swap

### PR DESCRIPTION
**Pull request by tcharding**
Currently a database Swap object has only a swap_id.  Add all the fields from Metadata which are used when displaying swap information to a client i.e., everything used by `with_swap_types!`.  Heavily clean up the unit tests.

This PR is not directly linked to an issue (I closed the (un-groomed) issue https://github.com/comit-network/comit-rs/issues/1423) that was an attempt to describe this problem since the problem is a not easily defined since we still do not know _exactly_ what has to be written to disk.  This PR adds data that is used by the `with_swap_types` macro.

Part of the Sqlite database Epic.

### CODE REVIEW

File: cnd/src/db/models.rs [hunk 1]
```diff
@@ -60,3 +70,101 @@ where
         Ok(SwapId(uuid))
     }
 }
+
+#[repr(i32)]
```
Comment (thomaseizinger) [inline cnd/src/db/models.rs]: `repr` is only used for alignment, I think we should not abuse this here. I know that the diesel docs do it as well but I think it is wrong.

Better add an explicit method that returns an integer for your variant if you need that:

```rust
impl Role {
   fn index(&self) -> u32 {
      match self {
          Role::Alice => 0,
          Role::Bob => 1,
	  }
   }
}
```

Comment (thomaseizinger) [reply cnd/src/db/models.rs]: However, as mentioned, I think they should be strings.

File: cnd/src/db/mod.rs [hunk 2]
```diff
@@ -159,158 +159,227 @@ pub fn default_db_path() -> Option<PathBuf> {
...
+        pub alpha_asset: AssetKind,
+        pub beta_asset: AssetKind,
+        pub role: Role,
     }
 
-    fn insertable_swap() -> InsertableSwap {
-        let id = example_swap_id();
-        create_insertable_swap(id)
+    impl Swap {
+        fn instantiate_test_case_1() -> Swap {
```
[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/mod.rs]: This stuff _is_ within `#[cfg(test)]`???

File: cnd/src/db/models.rs [hunk 3]
```diff
@@ -60,3 +70,101 @@ where
...
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[sql_type = "Integer"]
```
[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/models.rs]: Good points, will implement as suggested.

File: cnd/src/db/models.rs (hunk 1 above)
Comment (tcharding) [reply cnd/src/db/models.rs]: Cool, thanks for the review.

File: cnd/src/db/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply cnd/src/db/mod.rs]: Ah, my mistake sorry!
I thought this was the production code.
What do we need all this for though? :see_no_evil: 

As far as I can see, you only use them to instantiate test data and then convert to the `InsertableSwap`? Wouldn't functions that give you `InsertableSwap` instances be sufficient?

[the remark this answers is not in the document]
Comment (tcharding) [reply cnd/src/db/mod.rs]: Nice observation, fixed as suggested.

File: cnd/src/db/models.rs (hunk 1 above)
Comment (tcharding) [reply cnd/src/db/models.rs]: Implemented as suggested.  Discussed use of macro to generate the `to_sql` and `from_sql` trait implementations for enums.  Was only able to do `to_sql` - `from_sql` should be possible with some extra thought.  Leaving for another time.

Comment (tcharding) [reply cnd/src/db/models.rs]: If a way to do the macro jumps out at anyone feel free to say and I'll implement it.

File: cnd/src/db/models.rs [hunk 4]
```diff
@@ -6,57 +6,214 @@ use diesel::{
     sql_types::Text,
     Insertable, Queryable, *,
 };
-use std::{io::Write, str::FromStr, string::ToString};
+use failure::_core::{
```
Comment (luckysori) [inline cnd/src/db/models.rs]: :mega: `Debug`, `Display` and `Deref` should come from the standard library. `failure` is just re-exporting them here.

File: cnd/src/db/mod.rs [hunk 5]
```diff
@@ -159,158 +159,188 @@ pub fn default_db_path() -> Option<PathBuf> {
...
     #[test]
-    fn can_count_empty_database() {
+    fn can_count_empty_db() {
         let db = new_temp_db();
-        assert_eq!(0, db.count().unwrap());
+        assert_eq!(0, db.count().expect("db count error"));
     }
 
     #[test]
     fn insert_return_vaule_is_as_expected() {
```
Comment (D4nte) [inline cnd/src/db/mod.rs]:
```suggestion
    fn given_insert_return_value_is_as_expected() {
```

File: cnd/src/db/mod.rs [hunk 6]
```diff
@@ -159,158 +159,188 @@ pub fn default_db_path() -> Option<PathBuf> {
...
 
-        let swap = insertable_swap();
-        let rows_inserted = db.insert(swap).unwrap();
+        let swap = instantiate_test_case_1();
+        let rows_inserted = db.insert(swap).expect("db insert error");
         assert_eq!(rows_inserted, 1);
     }
 
     #[test]
     fn count_works_afer_insert() {
```
Comment (D4nte) [inline cnd/src/db/mod.rs]:
```suggestion
    fn count_works_after_insert() {
```

File: cnd/src/db/mod.rs [hunk 7]
```diff
@@ -159,158 +159,188 @@ pub fn default_db_path() -> Option<PathBuf> {
...
-        let _rows_inserted = db.insert(swap).unwrap();
-        assert_eq!(1, db.count().unwrap());
+        let swap = instantiate_test_case_1();
+        let _rows_inserted = db.insert(swap).expect("db insert error");
+
+        assert_eq!(1, db.count().expect("db count error"));
     }
 
     #[test]
     fn can_not_add_same_record_twice() {
```
Comment (D4nte) [inline cnd/src/db/mod.rs]:
```suggestion
    fn cannot_add_same_record_twice() {
```

File: cnd/src/db/models.rs (hunk 4 above)
Comment (D4nte) [reply cnd/src/db/models.rs]: Also, they are always in scope, no need to import them at all.

**Pull request merged**